app/main.py

```diff
@@ -1,34 +1,110 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    """A class to create ship deck."""
+
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        """Constructs all the necessary attributes for Deck object."""
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
+        self.deck = row, column
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    """A class to create a ship."""
+
+    def __init__(
+            self,
+            start: tuple[int, int],
+            end: tuple[int, int],
+            is_drowned: bool = False
+    ) -> None:
+        """Create decks and save them to a list `self.decks`"""
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = self.create_ship()
+
+    def create_ship(self) -> list[Deck]:
```

Comment:
return is optional here, import Optional from typing
```suggestion
    def create_ship(self) -> Optional[List[Deck]]:
```